model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -489,33 +490,66 @@ message TestStreamPayload {
   // (Required) the coder for elements in the TestStream events
   string coder_id = 1;
 
+  // (Optional) If specified, the TestStream will replay these events.
   repeated Event events = 2;
 
+  // (Optional) If specified, the TestStream will use the InteractiveService
+  // protocol found in beam_interactive_api.proto to retrieve events.
+  ApiServiceDescriptor endpoint = 3;
+
   message Event {
     oneof event {
       AdvanceWatermark watermark_event = 1;
       AdvanceProcessingTime processing_time_event = 2;
       AddElements element_event = 3;
     }
 
+    // Advances the watermark to the specified timestamp.
     message AdvanceWatermark {
+      // (Required) The watermark to advance to.
       int64 new_watermark = 1;
+
+      // (Optional) The output watermark tag for a PCollection. If unspecified
```

Comment:
I believe this should be required instead of having SDKs specially handle this logic.